### ISSUE TITLE
Android mobile app fails to load chromecast devices after a disconnect.

### ISSUE THREAD

**Issue by bengalih** [jellyfin/jellyfin#1753]
**Describe the bug**
After disconnecting from a chromecast device via the mobile app, you are unable to connect to (or even enumerate) any other chromecast devices.

**To Reproduce**
1. Press the cast button and choose "Chromecast" from the list
2. Observe enumerated chromecast devices and choose one to connect.
3. After connected, choose to disconnect.
4. Answer yes when asked "Would you like to shut down jellyfin on XXX speaker?"
5. Repeat steps 1 and 2

**Expected behavior**
You would expect that the Chromecast list would enumerate devices again to choose from, but absolutely nothing happens when you press the button on subsequent tries.

Only solution I have found is to forcibly stop the app via android settings and then relaunch.
Doing this however also introduces another bug which is that it will now list devices multiple times (appears to list the device you were previously connected to more than once).

**Logs**
don't know how to gather logs on mobile app

**Screenshots**
https://i.postimg.cc/85K2R31m/Screenshot-2019-09-13-11-40-47.png
https://i.postimg.cc/DwktP1KH/Screenshot-2019-09-13-11-40-55.png
https://i.postimg.cc/CLf38m9b/Screenshot-2019-09-13-11-41-05.png
https://i.postimg.cc/MHgg4Qfc/Screenshot-2019-09-13-11-41-15.png
https://i.postimg.cc/JzNS6trn/Screenshot-2019-09-13-11-41-31.png
https://i.postimg.cc/pdk4v7nM/Screenshot-2019-09-13-11-41-37.png

**System (please complete the following information):**
 - OS: Android, multiple versions including 7 and 10.
 - Browser: n/a
 - Jellyfin Version: 10.3.7
 - Reverse proxy: nginx, but this is reproduced via direct access on local network.

**Additional context**

**Comment by xplatinum** [comment]
I'm having this problem too.

**Comment by anthonylavado** [comment]
This may be related to a bug in the currently shipping app that we are aware about. There are some other fixes that we still need to land before we ship the fixed app.

**Comment by anthonylavado** [comment]
Yeah, so this is related to https://github.com/jellyfin/jellyfin-android/issues/176 which is getting fixed tonight.

**Comment by anthonylavado** [comment]
Hey there, the new version of the app is up. Can you try it and let me know if you still have the same issue? L
Thanks!

**Comment by bengalih** [comment]
> Hey there, the new version of the app is up. Can you try it and let me know if you still have the same issue? L
> Thanks!

Hey Anthony.  The issue is quite a bit improved, but there are still issues.  The main issue described seems to be fixed, so if you say "yes" when asked in Step #4 above, it will stop playing on the speaker and then you will  be able to select another CC device from the cast menu.

However, if you choose "no" when asked in Step #4 then the devices will fail to enumerate as described in the OP (i.e. nothing will show up under Chromecast, as if you failed to press it).

It also appears though that when relaunching the app, there are not duplicates of speakers shown, so that is a good sign.

I have also noticed another bug around this which is that if you connect to Speaker A, shutdown saying "Yes" in Step #4, then connect to Speaker B, when you go to stop playing from Speaker A it asks you in step #4 about Speaker A.  In other words, it is always asking about the speaker you stopped playing on one *prior* to the one you are currently playing on.

My specific case scenario is that I need to start playing on Speaker A, then disconnect from that speaker in the app, but I have to choose "no" from the shutdown dialog because I want it to keep playing on that speaker.  However I need to disconnect to then begin playing something else on Speaker B.  In practice this is me trying to play two different pieces of media on my kid's speakers at bedtime.

Let me know if you need any other tests/info/screen shots etc.

**Comment by anthonylavado** [comment]
@bengalih I don’t think we’ve considered this scenario at all before - do you have an example of an app that lets you easily do this? Something we can look to emulate?

I’ll open issues with the right places.

**Comment by bengalih** [comment]
> @bengalih I don’t think we’ve considered this scenario at all before - do you have an example of an app that lets you easily do this? Something we can look to emulate?
> 
> I’ll open issues with the right places.

So actually...no.  I was a Plex user prior (and still mainly am, but trying to stick with you guys as well as things improve).  With Plex, when you choose to disconnect from what you are casting you have two choices:  either do a straight stop, which stops playback entirely, or tell it to continue to playback on the device you are on (which in my case would be my phone from which I mainly cast to the other speakers from).  To get this to work in Plex android app, I would force close it - then I would be able to connect to another speaker.  In Plex's web interface if I close down the open browser/tab then reopen I can connect to another device while still playing on the other.

I'm sure both Plex and Jellyfin have to think that people are going to probably want to cast to more than one device simultaneously and that being said there might be one person in control of all that casting (especially in the case of minor children in an environment).

When I moved to JF I was excited by the option I saw in the mobile app because it seemed to somewhat work.  Let me ask you what is supposed to be the intent of the following dialog when disconnecting from a cc device?

"Would you like to shut down jellyfin on Chromecast"

Because in my tests, it does exactly what it sounds like ti should and exactly what I want.  Namely it disconnects my app from the cc device (since I chose "disconnect" in the app), but it does not shutdown jellyfin on my cc device and thus it continues to stream.
The only issue here is that if I choose to reconnect to another CC device, the app doesn't do anything.  So, what is it supposed to do?  It seems like you are 99% of the way there from providing the flexibility I need, but something isn't re-initializing properly after the disconnect.
